### DISCUSSION TITLE
How to use React SVG with sprite with Joint.js shape xlinkHref

### DISCUSSION THREAD

**Discussion by Baalakay** [clientIO/joint#2539] [Q&A]
### Introduction

I want to use the React native SVG rendering capability with an SVG sprite as input to the Joint.js shape image:xlinkHref.  React returns the correct image using the getImage() function below, but Joint.JS does not.  The output of React's xlinkHref is not a string SVG definition, but a React pointer to the individual 'dog' SVG like this.  

/static/media/sprite.d5c2218c941c0589a4791b8bbda46fc2.svg#dog

Any ideas how to make this work?

```
import icon from "./assets/icons/sprite.svg";

function getImage(id) {
    //temp set an image for testing
    return (
      <svg>
        <use xlinkHref={`${icon}#${id}`} />
      </svg>
    );
  }

  const dog = getImage().props.children.props.xlinkHref;

  const standardImage = new shapes.standard.Image({
    attrs: {
      image: {
        href: `${getImage('dog')}`,
      },
    },
  });
```

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS

**Comment by kumilingus** [comment]
It should work the same way as here: https://codepen.io/jointjs/pen/gOEqqQW

Did you miss to set the size of the image element?
```js
  const standardImage = new shapes.standard.Image({
    size: { width: 100, height: 100 },
    attrs: {
      image: {
        href: `${getImage('dog')}`,
      },
    },
  });
  ```

If that's not the case, please setup a functional example that could be debugged.

**Comment by Baalakay** [comment]
Hi Kumilingus - thanks for the reply.  Still does not work. Here is a [codesandbox.io link](https://codesandbox.io/p/devbox/baalakay-node-react-qwycl9?file=%2Fsrc%2FApp.js%3A13%2C18) to it. It works when the SVG is a variable, but not when pulled from an svg file (even without using a sprite). You will see the working image:href code commented out.

**Reply by kumilingus** [reply]
It works when:
- move the SVG file to `public` folder and use `href = '/bucket.svg'` 
- keep SVG file under `src` folder and import it with `import bucket from './bucket.svg'` and  use it with `href = bucket` ([example](https://codesandbox.io/p/devbox/baalakay-node-react-forked-2hhcy5?file=%2Fsrc%2FApp.js%3A19%2C8))

**Comment by Baalakay** [comment]
So in individual SVG file works now-- thanks! However, I still cannot get a sprite to render a specific element. New [example](https://codesandbox.io/p/devbox/baalakay-node-react-forked-7ncchs?file=%2Fsrc%2FApp.js%3A18%2C19) to show the issue. It works when using the sprite as a link, but as an imported file it displays all elements of the sprite.

**Reply by kumilingus** [reply]
These are not the same SVG files.

The one that you import as a file is missing the following styles:
```svg
  <defs>
    <style><![CDATA[
      .sprite { display: none; }
      .sprite:target { display: inline; }
    ]]></style>
  </defs>
```

See https://www.sitepoint.com/use-svg-image-sprites/#svgspritestacks.

**Comment by Baalakay** [comment]
Yeah, I tried with that included before but it doesn't compile, so I took it out. 

```
ERROR in ./src/sprite.svg
Module build failed (from ./node_modules/@svgr/webpack/lib/index.js):
Error: Expected node, got `
      .sprite { display: none; }
      .sprite:target { display: inline; }
    `
```

**Comment by Baalakay** [comment]
It works when I take out the ```<![CDATA[...]]>```.  I'll have to do some more studying up on sprite styling. Thanks so much for your help!

`  <defs>
    <style>
      .sprite { display: none; }
      .sprite:target { display: inline; }
    </style>
  </defs>`
